react/tic-tac-toe/src/app/screens/Login/index.js

```diff
@@ -1,44 +1,57 @@
 import React from 'react';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
 import { Field, reduxForm } from 'redux-form';
-import { func } from 'prop-types';
 
+import authActions from '~redux/auth/actions'; // eslint-disable-line import/no-unresolved
+
+import { validate } from './utils';
+import { VALIDATIONS, ERROR_MESSAGES } from './constants';
 import styles from './styles.module.scss';
-import { VALIDATIONS } from './constants';
 
 import Input from '~components/Input'; // eslint-disable-line import/no-unresolved
 
-function Login({ handleSubmit }) {
+import Alert from '~components/Alert'; // eslint-disable-line import/no-unresolved
+
+
+function Login({ handleSubmit, loginError }) {
   return (
     <div className={styles.loginWrapper}>
-      <div className={styles.loginInside}>
-        <form className={styles.loginForm} onSubmit={handleSubmit}>
-          <Field
-            name="email"
-            type="email"
-            component={Input}
-            label="Email"
-            placeholder="e.g. john.doe@email.com"
-            validate={VALIDATIONS.email}
-          />
-          <Field
-            name="password"
-            type="password"
-            component={Input}
-            label="Password"
-            placeholder="e.g. * * * * * * * *"
-            validate={VALIDATIONS.password}
-          />
-          <button className={styles.button} type="submit">
-              Sign In
-          </button>
-        </form>
-      </div>
+      <form className={styles.loginForm} onSubmit={handleSubmit}>
+        <Field
+          name="email"
+          type="email"
+          component={Input}
+          label="Email"
+          placeholder="e.g. john.doe@email.com"
+          validate={VALIDATIONS.email}
+        />
+        <Field
+          name="password"
+          type="password"
+          component={Input}
+          label="Password"
+          placeholder="e.g. * * * * * * * *"
+          validate={VALIDATIONS.password}
+        />
+        {loginError && (<Alert {...ERROR_MESSAGES[loginError]} />)}
+        <button className={styles.button} type="submit">
+          Sign In
+        </button>
+      </form>
     </div>
   );
 }
 
-Login.propTypes = {
-  handleSubmit: func.isRequired
-};
+const mapStateToProps = ({ auth }) => ({
```

Comment:
Use `store` here. Normally the `mapStateToProps` tend to grow quickly and the destructuring isn't very scalable. Also, using store makes the path much more readable.
`store.auth.loginError`

Reply:
Nice! I gonna have this in count.